src/main/java/school/lemon/changerequest/java/introduction/hw2/Task5.java

```diff
@@ -2,6 +2,10 @@
 
 public class Task5 {
     public static boolean commondEnd(int[] a, int[] b) {
+
+        if(a[0]==b[0] || a[a.length - 1] == b[b.length - 1] ||
```

Comment:
It's to hard to debug. You'd better split checks and assign them to appropriate variables.